sdk/core/Azure.Core/tests/NextLinkOperationImplementationTests.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+using Azure.Core.TestFramework;
+using NUnit.Framework;
+
+namespace Azure.Core.Tests
+{
+    public class NextLinkOperationImplementationTests
+    {
+        [Test]
+        public void ConstructRehydrationToken()
+        {
+            var requetMethod = RequestMethod.Get;
+            var startRequestUri = new Uri("https://test");
+            var nextRequestUri = "nextRequestUri";
+            var headerSource = "None";
+            string lastKnownLocation = null;
+            var finalStateVia = OperationFinalStateVia.OperationLocation.ToString();
+            var token = NextLinkOperationImplementation.GetRehydrationToken(requetMethod, startRequestUri, nextRequestUri, headerSource, lastKnownLocation, finalStateVia);
+            Assert.AreEqual(requetMethod, token.RequestMethod);
+            Assert.AreEqual(startRequestUri, token.InitialUri);
+            Assert.AreEqual(nextRequestUri, token.NextRequestUri);
+            Assert.AreEqual(headerSource, token.HeaderSource);
+            Assert.AreEqual(lastKnownLocation, token.LastKnownLocation);
+            Assert.AreEqual(finalStateVia, token.FinalStateVia);
+        }
+
+        [Test]
+        public void CreateWithIncompleteNextRequestUri()
+        {
+            var pipeline = CreateMockHttpPipeline();
+            var operationId = Guid.NewGuid().ToString();
+            var requestMethod = RequestMethod.Delete;
+            var rehydrationToken = new RehydrationToken(null, null, "None", $"/subscriptions/subscription-id/providers/Microsoft.Compute/locations/region/operations/{operationId}?api-version=2019-12-01", "https://test", requestMethod, null, OperationFinalStateVia.AzureAsyncOperation.ToString());
+            var operation = (NextLinkOperationImplementation)NextLinkOperationImplementation.Create(pipeline, rehydrationToken);
+            Assert.NotNull(operation);
+            Assert.AreEqual(operationId, operation.OperationId);
+            Assert.AreEqual(requestMethod, operation.RequestMethod);
+        }
+
+        [Test]
+        public void ConstructNextLinkOperation()
```

Comment:
Given the amount of logic around API version overrides and such, we should have test coverage here.

Reply:
API version input removed